### PR TITLE
mkcloud-dirmaint: Prepare for deploying muliple enviroments

### PR DESCRIPTION
**Pull request by rhafer**
This removes some hardcoded things from the dirmaint driver. And adds
code to "_cleanup" and "_prepare" to allow for multiple clouds being
deployed from a single host.

It will now setup automatically set a vswitch for the cloud, plug a nic
of the host into it and configure dnsmasq for the respective network.

However this code currently assumes that $cloud on z/VM will always
match ```mkcl<single-hex-digit>``` as it derives the mac addresses and vdev
names from that <single-hex-digit>.

### CODE REVIEW

File: scripts/lib/mkcloud-dirmaint.sh [hunk 1]
```diff
@@ -69,7 +105,7 @@ function _dirmaint_link_and_write_disk()
     # FIXME kill the machine
     # vmcp force $ruser || :
     # FIXME
-    local vdev="a100"
+    local vdev="0100"
```
Comment (dirkmueller) [inline scripts/lib/mkcloud-dirmaint.sh]: uh, why that? we don't want to replace mkcldhsts 0100 disk!

File: scripts/lib/mkcloud-dirmaint.sh [hunk 2]
```diff
@@ -30,16 +28,54 @@ function dirmaint_do_cleanup()
 
     killproc -p /var/run/mkcloud/dnsmasq-$cloud.pid /usr/sbin/dnsmasq
     rm -f /var/run/mkcloud/dnsmasq-$cloud.pid /etc/dnsmasq-$cloud.conf
+
+    # This assumes $cloud is named "mkcl<single-hex-digit>"
```
Comment (dirkmueller) [inline scripts/lib/mkcloud-dirmaint.sh]: can we add that to dirmaint_sanity_checks?

Comment (rhafer) [inline scripts/lib/mkcloud-dirmaint.sh]: Done

File: scripts/lib/mkcloud-dirmaint.sh (hunk 1 above)
Comment (rhafer) [inline scripts/lib/mkcloud-dirmaint.sh]: Ugh. I've changed that to be ```0a<idx>0``` now.

**Pull request merged**